gtsfm/bundle/bundle_adjustment.py

```diff
@@ -5,37 +5,147 @@
 
 import numpy as np
 from pathlib import Path
-from typing import List, NamedTuple, Tuple
+from typing import List, Tuple
 
 import dask
-import gtsam
 from dask.delayed import Delayed
-from gtsam import GeneralSFMFactorCal3Bundler, SfmTrack, Values, symbol_shorthand
+from gtsam import (
+    GeneralSFMFactor2Cal3Bundler,
+    LevenbergMarquardtOptimizer,
+    LevenbergMarquardtParams,
+    NonlinearFactorGraph,
+    PinholeCameraCal3Bundler,
+    PriorFactorCal3Bundler,
+    PriorFactorPose3,
+    SfmTrack,
+    Values,
+    symbol_shorthand,
+)
+from gtsam.noiseModel import Isotropic
 
 import gtsfm.utils.logger as logger_utils
 from gtsfm.common.gtsfm_data import GtsfmData
 from gtsfm.evaluation.metrics import GtsfmMetric, GtsfmMetricsGroup
 
 METRICS_PATH = Path(__file__).resolve().parent.parent.parent / "result_metrics"
 
+"""In this file, we use the GTSAM's GeneralSFMFactor2 instead of GeneralSFMFactor because Factor2 enables decoupling
+of the camera pose and the camera intrinsics, and hence gives an option to share the intrinsics between cameras.
+"""
+
 # TODO: any way this goes away?
-C = symbol_shorthand.C
-P = symbol_shorthand.P
+P = symbol_shorthand.P  # 3d point
```

Comment:
just curious, why the change in variable name here from C to X?

Reply:
C is for camera. I thought of using X for pose, as http://www.cs.cmu.edu/~kaess/pub/Dellaert17fnt.pdf uses *X* to describe robot locations [Fig 1.1]